datasetinsights/io/source/base.py

```diff
@@ -0,0 +1,40 @@
+import re
+from abc import ABCMeta, abstractmethod
+
+
+class DownloaderRegistry(ABCMeta):
+    registry = {}
+
+    def __new__(cls, name, bases, namespace):
+        protocal = "PROTOCAL"
```

Comment:
Is this supposed to be `PROTOCOL`? Or `PROTOCAL` means something else?

Reply:
This is a typo good catch